ApollosProject/apollos-prototype
# Adds Auth Module (and sneak peak of Schema Module)

## DESCRIPTION

### What does this PR do, or why is it needed?
- Moves auth into it's own data-connector module. 
- Creates data-schema module (currently _only_) used in auth module, haven't applied the changes to the main API branch, but will do that in a separate PR. 

### What design trade-offs/decisions were made?
- Chose to create the data-schema module, but not use it for the code in `apollos-church-api`. I will make that change when I transition the modules out from `apollos-church-api`.

### How do I test this PR?
1. Boot app.
2. Testing logging in, ensure you can still log in.
3. Test changing profile fields, ensure fields are actually changed.
4. Test logging out, ensure you can still log out.

### What automated tests did you write?
- Kept existing tests

## TODO:

- [x] Closes #404 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
